src/runtime/runtime_stm32wle5.go

```diff
@@ -4,21 +4,125 @@
 package runtime
 
 import (
+	"device/stm32"
 	"machine"
 )
 
 type arrtype = uint32
 
 func init() {
-	// Currently the clock is not configured, which means
-	// the MCU runs at default reset clock speed (4Mhz).
-	// Code to initialize RCC and PLL can go here.
+
+	// Configure 48Mhz clock
+	initCLK()
 
 	// UART init
 	machine.Serial.Configure(machine.UARTConfig{})
 
 	// Timers init
 	initTickTimer(&machine.TIM1)
+
+	// Radio init
+	subGhzInit()
+}
+
+func initCLK() {
+
+	// Set Power Voltage Regulator Range 2
+	//stm32.PWR.CR1.ReplaceBits(0b10, stm32.PWR_CR1_VOS_Msk, stm32.PWR_CR1_VOS_Pos)
+
+	// Enable HSE32 VDDTCXO output on package pin PB0-VDDTCXO
+	stm32.RCC.CR.ReplaceBits(stm32.RCC_CR_HSEBYPPWR_VDDTCXO, 0b1, stm32.RCC_CR_HSEBYPPWR_Pos)
+
+	// SYSCLOCK from HSE32 clock division factor (SYSCLOCK=HSE32)
+	stm32.RCC.CR.ReplaceBits(stm32.RCC_CR_HSEPRE_Div1, 0b1, stm32.RCC_CR_HSEPRE_Pos)
+
+	// enable external Clock HSE32 TXCO (RM0461p226)
+	stm32.RCC.CR.SetBits(stm32.RCC_CR_HSEBYPPWR)
+	stm32.RCC.CR.SetBits(stm32.RCC_CR_HSEON)
+	for !stm32.RCC.CR.HasBits(stm32.RCC_CR_HSERDY) {
+	}
+
+	// Disable PLL
+	stm32.RCC.CR.ClearBits(stm32.RCC_CR_PLLON)
+	for stm32.RCC.CR.HasBits(stm32.RCC_CR_PLLRDY) {
+	}
+
+	// Configure PLL
+	stm32.RCC.PLLCFGR.Set(0x23020613)
+
+	// Enable PLL
+	stm32.RCC.CR.SetBits(stm32.RCC_CR_PLLON)
+	for !stm32.RCC.CR.HasBits(stm32.RCC_CR_PLLRDY) {
+	}
+
+	// Enable PLL System Clock output.
+	stm32.RCC.PLLCFGR.SetBits(stm32.RCC_PLLCFGR_PLLREN)
+	for !stm32.RCC.CR.HasBits(stm32.RCC_CR_PLLRDY) {
+	}
+
+	// Set Flash Latency of 2 and wait until it's set properly
+	stm32.FLASH.ACR.ReplaceBits(0b010, 0b111, stm32.Flash_ACR_LATENCY_Pos)
+	for (stm32.FLASH.ACR.Get() & 0b11) != 0x2 {
+	}
+
+	// HCLK1 clock division factor (SYSCLK/512)
+	stm32.RCC.CFGR.ReplaceBits(0x0000, 0b1111, stm32.RCC_CFGR_HPRE_Pos)
+	for !stm32.RCC.CFGR.HasBits(stm32.RCC_CFGR_HPREF) {
+	}
+
+	// HCLK3 clock division factor (SYSCLK/512)
+	stm32.RCC.EXTCFGR.ReplaceBits(0x0000, 0b1111, stm32.RCC_EXTCFGR_SHDHPRE_Pos)
+	for !stm32.RCC.EXTCFGR.HasBits(stm32.RCC_EXTCFGR_SHDHPREF) {
+	}
+
+	// PCLK1 clock division factor (HCLK1/16)
+	stm32.RCC.CFGR.ReplaceBits(0x000, 0b111, stm32.RCC_CFGR_PPRE1_Pos)
+	for !stm32.RCC.CFGR.HasBits(stm32.RCC_CFGR_PPRE1F) {
+	}
+
+	// PCLK2 clock division factor (HCLK1/16)
+	stm32.RCC.CFGR.ReplaceBits(0x000, 0b111, stm32.RCC_CFGR_PPRE2_Pos)
+	for !stm32.RCC.CFGR.HasBits(stm32.RCC_CFGR_PPRE2F) {
+	}
+
+	// Set clock source to PLL (0x3)
+	stm32.RCC.CFGR.ReplaceBits(0b11, 0b11, stm32.RCC_CFGR_SW_Pos)
+	for (stm32.RCC.CFGR.Get() & stm32.RCC_CFGR_SWS_Msk) != 0xc {
+	}
+
+}
+
+// SubGhzInit() enables SX126x radio module
```

Comment:
I don't think this should be in the runtime package - runtime is really about setting up clocks and time.

Should probably think about this as an optional peripheral that apps may (or may not) use - with it either:
1 - being in machine package
2 - being supported in the tinygo drivers repo

I'm not sure which is best - it's not a SPI/I2C peripheral - and might be a bit 'specialized' for the machine package?  Perhaps @aykevl  or @deadprogram  have some thoughts?

Reply:
SX126x Radio (which is part of STM32WL) will have its own tinygo driver soon, so the code can be reused when using SX126x standalone module.
 
But this function is specific to STM32WL series (as SX126x radio unit and MCU are on the same die)
 
IMHO it's entirely part of the whole chipset initialisation process.

I'm not sure moving this function to tinygo-driver makes sense (I don't see other situations you might need this)

But having it in machine package instead, would not hurt. 

Other comments welcome .

Reply:
`machine` package is a good place for this. Perhaps it should be exported so users who want to turn on this functionality can do so, but it is not always turned on by default?

Reply:
SubGhzInit function moved in "src/machine/machine_stm32wle5.go" and exported so users can use it or not